server/src/search.rs

```diff
@@ -127,24 +135,37 @@ struct MSFacetDistribution {
     facet: HashMap<String, i32>,
 }
 
-pub async fn do_benchmarked_search(q: String, args: SearchQueryArgs) -> Result<SearchResults> {
+pub async fn do_benchmarked_search(q: String, args: SearchQueryArgs) -> SearchResults {
     let start_time = Instant::now();
 
     let results_sections = execute_search(q, args).await;
 
     let time_ms = start_time.elapsed().as_millis();
-    Ok(SearchResults {
+    SearchResults {
         sections: results_sections,
         time_ms,
-    })
+    }
 }
 
 // size=100 seems to be about 10M
 #[cached(size = 500)]
 async fn execute_search(q: String, args: SearchQueryArgs) -> Vec<SearchResultsSection> {
     let parsed = parse_input_query(q);
+    let sanatised_args = sanitise_args(args);
+    return do_geoentry_search(&parsed.tokens, sanatised_args).await;
+}
 
-    return do_geoentry_search(&parsed.tokens, args).await;
+fn clamp_limit(val: usize, min_val: usize) -> usize {
+    // Clamp a value to min_val..1000
+    max(min_val, min(1_000, val))
+}
+
+fn sanitise_args(args: SearchQueryArgs) -> SanitisedSearchQueryArgs {
+    SanitisedSearchQueryArgs {
+        limit_buildings: clamp_limit(args.limit_buildings.unwrap_or(5), 0),
+        limit_rooms: clamp_limit(args.limit_rooms.unwrap_or(5), 0),
+        limit_all: clamp_limit(args.limit_all.unwrap_or(20), 1),
```

Comment:
```suggestion

fn sanitise_args(args: SearchQueryArgs) -> SanitisedSearchQueryArgs {
    SanitisedSearchQueryArgs {
        limit_buildings: args.limit_buildings.unwrap_or(5).clamp(0, 1_000),
        limit_rooms: args.limit_rooms.unwrap_or(5).clamp(0, 1_000),
        limit_all: args.limit_all.unwrap_or(20).clamp(1, 1_000),
```
We can use Rust's built-in clamp here.

Reply:
Ah, nice to know xD